core/dispatcher/src/whisk/core/invoker/Invoker.scala

```diff
@@ -163,15 +166,20 @@ class Invoker(
         // when records are fetched, invoke action
         val activationDocFuture =
             actionFuture flatMap { theAction =>
+                // assume this future is done here
+                marker(this, INVOKER_FETCH_ACTION_DONE)
```

Comment:
in failure of the fetch futures, you will need to emit the appropriate end marker as well.